Header enum_header
string state
